crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -300,106 +300,79 @@ pub struct EncryptionInfo {
 /// Represents a matrix room event that has been returned from `/sync`,
 /// after initial processing.
 ///
-/// This is almost identical to [`TimelineEvent`], but wraps an
-/// [`AnySyncTimelineEvent`] instead of [`AnyTimelineEvent`].
-#[derive(Clone, Deserialize, Serialize)]
+/// Previously, this differed from [`TimelineEvent`] by wrapping an
+/// [`AnySyncTimelineEvent`] instead of an [`AnyTimelineEvent`], but nowadays
+/// they are essentially identical, and one of them should probably be removed.
+#[derive(Clone, Serialize)]
 pub struct SyncTimelineEvent {
-    /// The actual event.
-    #[serde(rename = "event")]
-    inner_event: Raw<AnySyncTimelineEvent>,
-
-    /// The encryption info about the event. Will be `None` if the event was not
-    /// encrypted.
-    #[serde(rename = "encryption_info")]
-    inner_encryption_info: Option<EncryptionInfo>,
+    /// The event itself, together with any information on decryption.
+    inner: TimelineEventInner,
 
     /// The push actions associated with this event.
-    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
     pub push_actions: Vec<Action>,
-    /// The encryption info about the events bundled in the `unsigned` object.
-    ///
-    /// Will be `None` if no bundled event was encrypted.
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub unsigned_encryption_info: Option<BTreeMap<UnsignedEventLocation, UnsignedDecryptionResult>>,
 }
 
 impl SyncTimelineEvent {
     /// Create a new `SyncTimelineEvent` from the given raw event.
     ///
-    /// This is a convenience constructor for when you don't need to set
-    /// `encryption_info` or `push_action`, for example inside a test.
+    /// This is a convenience constructor for a plaintext event when you don't
+    /// need to set `push_action`, for example inside a test.
     pub fn new(event: Raw<AnySyncTimelineEvent>) -> Self {
-        Self {
-            inner_event: event,
-            inner_encryption_info: None,
-            push_actions: vec![],
-            unsigned_encryption_info: None,
-        }
+        Self { inner: TimelineEventInner::PlainText { event }, push_actions: vec![] }
     }
 
     /// Create a new `SyncTimelineEvent` from the given raw event and push
     /// actions.
     ///
-    /// This is a convenience constructor for when you don't need to set
-    /// `encryption_info`, for example inside a test.
+    /// This is a convenience constructor for a plaintext event, for example
+    /// inside a test.
     pub fn new_with_push_actions(
         event: Raw<AnySyncTimelineEvent>,
         push_actions: Vec<Action>,
     ) -> Self {
-        Self {
-            inner_event: event,
-            inner_encryption_info: None,
-            push_actions,
-            unsigned_encryption_info: None,
-        }
+        Self { inner: TimelineEventInner::PlainText { event }, push_actions }
```

Comment:
Are we sure it's only used for plain text events, even in testing?

Reply:
Well, the fact is that the old code was setting `inner_encryption_info` (previously `encryption_info`) to `None`, which is pretty much the definition of a plaintext event, or at least one we're choosing to treat as such (which currently includes events we failed to decrypt).

Part of our problem here is that, for now, we're using `PlainText` to represent both actual plaintext events, and also events that we failed to decrypt. I think that's a fair representation of the way the code currently works, but does lead to a slightly confusing situation.

I'm working on a followup PR which will add a third entry to the `TimelineEventKind` enum, and we might need to figure out which calls to these methods actually want `PlainText`, and which want `UnableToDecrypt` then.